src/sinol_make/commands/run/__init__.py

```diff
@@ -615,21 +691,29 @@ def run(self, args):
 			'java_compiler_path': args.java_compiler_path
 		}
 
-		if 'oiejq_path' in args and args.oiejq_path is not None:
-			if not util.check_oiejq(args.oiejq_path):
-				util.exit_with_error('Invalid oiejq path.')
-			self.timetool_path = args.oiejq_path
-		else:
-			self.timetool_path = util.get_oiejq_path()
-		if self.timetool_path is None:
-			util.exit_with_error('oiejq is not installed.')
+		if args.time_tool == 'oiejq':
+			if sys.platform != 'linux':
+				util.exit_with_error('oiejq is only available on Linux.')
+			if 'oiejq_path' in args and args.oiejq_path is not None:
+				if not util.check_oiejq(args.oiejq_path):
+					util.exit_with_error('Invalid oiejq path.')
+				self.timetool_path = args.oiejq_path
+			else:
+				self.timetool_path = util.get_oiejq_path()
+			if self.timetool_path is None:
+				util.exit_with_error('oiejq is not installed.')
+		elif args.time_tool == 'time':
+			if sys.platform == 'win32' or sys.platform == 'cygwin':
+				util.exit_with_error('Measuring with `time` is not supported on Windows.')
+			self.timetool_path = None
```

Comment:
I feel like those `if`s would be better placed somewhere where the command-line arguments are handled. This logic doesn't really suit to a function named `print_expected_scores`, as it doesn't have to do with printing.

Reply:
Changed ba6e23795437d0426d93d39f52f4ce701eb219c4